src/lib.rs

```diff
@@ -60,98 +41,251 @@ fn set_stack_limit(l: usize) {
 ///
 /// The closure `f` is guaranteed to run on a stack with at least `red_zone`
 /// bytes, and it will be run on the current stack if there's space available.
+#[inline(always)]
 pub fn maybe_grow<R, F: FnOnce() -> R>(red_zone: usize,
                                        stack_size: usize,
                                        f: F) -> R {
     if remaining_stack() >= red_zone {
         f()
     } else {
-        grow_the_stack(stack_size, f)
+        grow(stack_size, f)
     }
 }
 
 /// Queries the amount of remaining stack as interpreted by this library.
 ///
 /// This function will return the amount of stack space left which will be used
 /// to determine whether a stack switch should be made or not.
+#[inline(always)]
 pub fn remaining_stack() -> usize {
-    unsafe {
-        __stacker_stack_pointer() - get_stack_limit()
-    }
+    &mut () as *mut _ as usize - get_stack_limit()
```

Comment:
I would personally prefer to stick with `__stacker_stack_pointer()`, this seems like the kind of thing that's changing all the time in the compiler to be optimized one way or another.

Reply:
I think this is a good idea too, because we don't know where the `()` will be place at the stack frame, there might be a large local below it giving an inaccurate stack pointer.